### PR TITLE
updated robot.e.registerIntegration and robot.e.create auth handling

### PR DESCRIPTION
**Pull request by mosheBelostotsky**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/eedevops/he-design/pull/20" target="_blank">ReviewNinja

File: hubot_enterprise.md [hunk 1]
```diff
@@ -137,22 +143,21 @@ Used by integration developers to register their integration with `hubot-enterpr
...
+    # check if integration+function have auth
     # pass auth to auth engine (use bluebird promises)
     return auth_engine.authenticated(userInfo, integration)
-    .then (res) ->
+    .then (auth_obj) ->
       # add secrets to msg object
-      msg.auth_secrets = res
-      callback(msg)
+      # not adding auth_obj on msg, so it would not be exposed to all code
+      callback(msg, auth_obj)
```
Comment (mosheBelostotsky) [inline hubot_enterprise.md]: is there a way for the callback to **revoke** the auth tokens?
e.g: when finding out that tokens are expired during a call?

File: hubot_enterprise.md [hunk 2]
```diff
@@ -22,10 +22,16 @@ Used by integration developers to register their integration with `hubot-enterpr
   - `metadata`: Metadata object to be stored and displayed in help
     - `short_desc`: short description for integration
     - `long_desc`: long description for integration (allow .md files loading here?)
-  - `authentication`: authentication object
-    - `type`: one of the allowed type in auth module, e.g: 'username_password'
-    - `options`:
-      - `callback`: callback function (if auth require this)
+    - `name`: integration **alias** to be called with
+  - `authentication`: authentication object - please refer to ??? for a list of available auth objects
```
Comment (mosheBelostotsky) [inline hubot_enterprise.md]: @rqc where can we find a list of currently supported auth objects and args to pass?

File: hubot_enterprise.md (hunk 1 above)
Comment (rqc) [inline hubot_enterprise.md]: @mosheBelostotsky we could provide a mechanism for _signing out_ of an integration for a particular user. We could discuss and plan this for later.

**Pull request merged**